articles/quickstart/spa/angular-beta/02-calling-an-api.md

```diff
@@ -0,0 +1,272 @@
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+ ```
+
+Next, open `src/app/external-api/external-api.component.ts` and replace its contents with the following:
+
+```js
+// src/app/external-api/external-api.component.ts
+
+import { Component } from '@angular/core';
+import { ApiService } from 'src/app/api.service';
+
+@Component({
+  selector: 'app-external-api',
+  templateUrl: './external-api.component.html',
+  styleUrls: ['./external-api.component.css']
+})
+export class ExternalApiComponent {
+  responseJson: string;
+  hasResponse = false;
+
+  constructor(private apiService: ApiService) {}
+
+  async pingApi() {
+    const response = await this.apiService.ping();
+
+    this.responseJson = JSON.stringify(response, null, 2).trim();
+    this.hasResponse = true;
+  }
+}
+
+```
+
+This component imports the `ApiService` that was created in the previous step, and calls the `ping()` method on it when the **Ping API** button is clicked. The response from this call is then saved into a local property, and then ultimately displayed on the screen in JSON format.
+
+With this in place, modify the application router so that this new page can be accessed. Open the `src/app/app-routing.module.ts` file and modify the routes list to include the new component for calling the API:
+
+```js
+// src/app/app-routing.module.ts
+
+// .. other imports
+
+// NEW - import the ExternalApiComponent class
+import { ExternalApiComponent } from './external-api/external-api.component';
+
+const routes: Routes = [
+  {
+    path: '',
+    component: HomeComponent
+  },
+  {
+    path: 'callback',
+    component: CallbackComponent
+  },
+  {
+    path: 'profile',
+    component: ProfileComponent,
+    canActivate: [AuthGuard]
+  },
+
+  // NEW - add a route to the External API component
+  {
+    path: 'external-api',
+    component: ExternalApiComponent,
+    canActivate: [AuthGuard]
+  }
+];
+```
+
+Note that the `external-api` route also makes use of the authentication guard. This prevents the route from being accessed by unauthenticated users.
+
+Finally, add a link to the navigation bar so that the user may reach this new page. Open `src/app/navbar/navbar.component.html` and modify it to include a link to the `/external-api` route:
+
+```html
+<header>
+  <button (click)="login()" *ngIf="!isAuthenticated">Log in</button>
+  <button (click)="logout()" *ngIf="isAuthenticated">Log out</button>
+
+  <!-- NEW - add a couple of router links to the home page, and to profile -->
+  <a [routerLink]="['']">Home</a>&nbsp;
+  <a [routerLink]="['profile']" *ngIf="isAuthenticated">Profile</a>&nbsp;
+  <a [routerLink]="['external-api']" *ngIf="isAuthenticated">External API</a>
+</header>
+```
+
+At this point, you should be able to run the application, browse to the External API page and click the **Ping API** button to initiate the API call. The results of that call should then be available in the browser.
```

Comment:
Mention that npm needs to be restarted here for the server changes to take hold.